core/src/subgraph/config.rs

```diff
@@ -0,0 +1,26 @@
+use graph::prelude::{regex::Regex, SubgraphName};
+
+#[derive(Default, Debug)]
+pub struct SubgraphPerfRules {
+    pub rules: Vec<(Regex, SubgraphPerfConfig)>,
+}
+
+impl SubgraphPerfRules {
+    pub fn config_for_name(&self, name: &SubgraphName) -> Option<SubgraphPerfConfig> {
+        self.rules.iter().find_map(|(regex, config)| {
+            if regex.is_match(name.as_str()) {
+                Some(config.clone())
```

Comment:
If this method returns a `Option<&SubgraphPerfConfig>`, we don't need to clone; the caller can always decide whether they need a clone or not.